core/arch/aarch64/emit_utils.c

```diff
@@ -730,13 +795,77 @@ append_save_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
                               opnd_create_reg(REG_DCXT),
                               OPND_CREATE_INTPTR(offsetof(priv_mcontext_t, simd))));
     for (i = 0; i < 32; i += 2) {
-        /* stp q(i), q(i + 1), [x1, #(i * 16)] */
+        /* stp q(i), q(i + 1), [x1, #(i * 16)]
+         * From the AArch64 manual:
+         * "The signed immediate byte offset is a multiple of 16 in the range
+         * -1024 to 1008, defaulting to 0 and encoded in the imm7 field as
+         * <imm>/16."
+         */
         APP(ilist,
             INSTR_CREATE_stp(
                 dcontext,
                 opnd_create_base_disp(DR_REG_X1, DR_REG_NULL, 0, i * 16, OPSZ_32),
                 opnd_create_reg(DR_REG_Q0 + i), opnd_create_reg(DR_REG_Q0 + i + 1)));
     }
+    if (proc_has_feature(FEATURE_SVE)) {
+        for (i = 0; i < 32; i++) {
+            /* str z(i), [x1, #(i mul vl)]
+             * "Store a vector register to a memory address generated by a
+             * 64-bit scalar base, plus an immediate offset in the range -256
+             * to 255 which is multiplied by the current vector register size
+             * in bytes."
+             */
+            APP(ilist,
+                INSTR_CREATE_str(
+                    dcontext,
+                    opnd_create_base_disp(
+                        DR_REG_X1, DR_REG_NULL, 0, i,
```

Comment:
Another displacement that looks incorrect; another below.

Reply:
See https://github.com/DynamoRIO/dynamorio/pull/5835#discussion_r1252174548 above.

Reply:
Fixed by SVE LDR/STR https://github.com/DynamoRIO/dynamorio/pull/6230, emitter and mangler updated.